Assignment 3.Rmd

```diff
@@ -127,27 +113,82 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+```{r}
+#The data with 3 clusters is more informative
+fit2 <- kmeans(K3,centers=3,nstart=20,iter.max=50)
+fit2$cluster
+K8<-data.frame(K3,fit2$cluster)
+names(K8)<-c("1","2","3","4","5","cluster")
+K9 <- K8%>%gather(week, motivation,1:5)%>%group_by(week,cluster)
+K10 <- K9%>%summarize(avg_motivation=mean(motivation))
+K10$week <- as.numeric(K10$week)
+K10$cluster <- as.factor(K10$cluster)
+ggplot(K10,aes(x=week,y=avg_motivation,col=cluster))+geom_line()+xlab("Week")+ylab("Average Motivation")
+```
 
-Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
+Once you have done this, save both of your plots to the Assignment 3 file. Create a Zotero item to index your new computer program (Assignment 3.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
 ##Part II
 
 Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up. 
 
+```{r}
+D1<-read.csv("HUDK405018-cluster.csv",header=T)
+D2<-D1[-(1:2),]
+D3<-select(D2,Q1_1:Q1_2,Q2_1:Q2_7)
+D3[D3==""] <- NA
+D4<-na.omit(D3)
+
+#Cluster by Region
+D5<-geocode(as.character(D4$Q1_1),output = "latlon",source = "dsk")
+D6<-scale(D5)
+fit3 <- kmeans(D6,centers=4,nstart=20,iter.max=50)
+fit3$cluster
+D7<-data.frame(D6,fit3$cluster)
+names(D7)<-c("longitude","latitude","cluster")
+D7$cluster <- as.factor(D7$cluster)
+ggplot(D7,aes(x=longitude,y=latitude,col=cluster))+geom_point()+xlab("Longitude")+ylab("Latitude")
+
+#Cluster by Answers to the Questions
+D8<-select(D4,3:9)
+D9<-scale(data.matrix(D8))
+fit4 <- kmeans(D9,centers=4,nstart=20,iter.max=50)
+fit4$cluster
+D10<-data.frame(D9,fit4$cluster)
+names(D10)<-c("1","2","3","4","5","6","7","cluster")
+D11 <- D10%>%gather(activity,hours,1:7)
+D12 <- D11%>%group_by(activity,cluster)%>%summarize(avg_hours=mean(hours))
+D12$activity <- as.numeric(D12$activity)
+D12$cluster <- as.factor(D12$cluster)
+ggplot(D12,aes(x=activity,y=avg_hours,col=cluster))+geom_line()+xlab("Activity")+ylab("Average Hours")
+```
+
+
 ##Part III
 
-Create a visualization that shows the overlap between the two groups of clusters you created in part III.
+Create a visualization that shows the overlap between the two groups of clusters you created in part II.
+```{r}
+D13<-data.frame(D7,D10)
+colnames(D13)[3] <- "cluster_region"
+colnames(D13)[11]<- "cluster_activity"
+D13$cluster_activity<- as.factor(D13$cluster_activity)
+ggplot(D13) + geom_point(aes(color= cluster_region, shape=cluster_activity,x=longitude, y=latitude) ) 
```

Comment:
Does this plot allow you to see where the overlap is between the two groups of clusters?